src/backend/window.rs

```diff
@@ -283,22 +289,35 @@ pub struct WindowUniforms {
 
 	/// The size of the image in pixels.
 	pub pixel_size: [f32; 2],
+
+	/// The zoom of the image.
+	pub zoom: [f32; 2],
+
+	/// The pan of the image.
+	pub pan: [f32; 2],
 }
 
 impl WindowUniforms {
 	pub fn no_image() -> Self {
-		Self::stretch([0.0; 2])
+		Self::stretch([0.0; 2], [1.0, 1.0], [0.0; 2])
 	}
 
-	pub fn stretch(pixel_size: [f32; 2]) -> Self {
+	pub fn stretch(pixel_size: [f32; 2], zoom: [f32; 2], pan: [f32; 2]) -> Self {
```

Comment:
Maybe we should implement a builder-like API for `WindowUniforms` instead of taking these values in each constructor. The zoom and pan can default to no-ops, and be set later with chainable setters like this:

```rust
pub fn set_zoom(zoom: [f64; 2]) -> Self {
    self.zoom = zoom;
    self
}
```